### DISCUSSION TITLE
Start Jamulus two times on two sound cards

### DISCUSSION THREAD

**Discussion by newlaurent62** [jamulussoftware/jamulus#1397] [Ideas]
Hello,

I'm on Linux (UbuntuStudio)

I created two presets for two sound cards in qjackctl : U22 and XR18.

With the commands 

`qjackctl -n U22 & qjackctl -n XR18 &`

I can start Jack two times with different parameters by deactivating the Service D-Bus.

But Jamulus stuck at finding the jack server. (There is no DEFAULT Jack server)

Could it be possible to add a command line parameter "Jack Server Name", to select the Jack server to use ?

BR,
Laurent

**Comment by djfun** [comment]
Should be possible with something like this: https://github.com/djfun/jamulus/commit/792381f9c2aa0b871e7e73895d89f4ac1b9a4cce

**Reply by newlaurent62** [reply]
Yes that is exactly what I needed.

**Reply by newlaurent62** [reply]
Thanks I will test if i can download your code.

**Comment by newlaurent62** [comment]
![Two Jamulus - Two jack server](https://user-images.githubusercontent.com/10225394/113099117-bf760880-91f9-11eb-9359-567c957fb47f.png)

**Reply by newlaurent62** [reply]
I started one jack server to default (U22) to be able to use audio applications and one jack server with another name (XR18). The audio player play music on Jamulus U22.

**Reply by djfun** [reply]
does it work?

**Reply by newlaurent62** [reply]
Yes i can hear the music on both sound cards thanks to Jamulus on a single computer. It compiles/runs and works.

**Reply by newlaurent62** [reply]
I used the following command line:

`qjackctl &
qjackctl -n XR18 &
./Jamulus --clientname U22 &
./Jamulus --jackservername XR18 --clientname XR18 &`

**Reply by hoffie** [reply]
@newlaurent62 Can you check whether vanilla Jamulus (i.e. the official 3.7.0) + using the `JACK_DEFAULT_SERVER` variable as described in https://github.com/jamulussoftware/jamulus/pull/1406#issuecomment-811222814 would also work for you?

**Reply by newlaurent62** [reply]
Yes it works fine with all jack applications.

`qjackctl -n XR18 & 
export JACK_DEFAULT_SERVER=XR18;Jamulus`

**Comment by djfun** [comment]
@hoffie How should we proceed in integrating this commit? Should I create a merge request? Or do we need a proper issue first?

**Reply by ann0see** [reply]
I think you can just open a Pull Request since the coda already exists.

**Reply by djfun** [reply]
See #1406

**Comment by djfun** [comment]
I prepared a new commit with a hint to the `JACK_DEFAULT_SERVER` env variable: https://github.com/djfun/jamulus/commit/44d9239cd4de093d642f06d5bd3b54ea7285a0b5

**Reply by hoffie** [reply]
Cool, want to open an PR for that? I'm hopeful that it can be merged :)

**Comment by dakhubgit** [comment]
In GNU/Linux, the sound card selector only offers the option "Jack" which is somewhat pointless.  One could let it either select one of several different Jack servers or let it specify the actual input (pair?) to be connected with Jamulus.

If it is the former, this would affect this discussion/MR I think.
